dags/sources/tasks/business_logic/db_write_suggestion.py

```diff
@@ -0,0 +1,95 @@
+import json
+import logging
+from datetime import datetime
+
+import pandas as pd
+from shared.tasks.database_logic.db_manager import PostgresConnectionManager
+from sources.config import shared_constants as constants
+
+logger = logging.getLogger(__name__)
+
+
+def db_write_suggestion(
+    dag_name: str,
+    run_id: str,
+    df_acteur_to_create: pd.DataFrame,
+    df_acteur_to_delete: pd.DataFrame,
+    df_acteur_to_update: pd.DataFrame,
+):
+
+    metadata = {}
+
+    run_name = run_id.replace("__", " - ")
+
+    insert_suggestion(
+        df=df_acteur_to_create,
+        metadata=metadata,
+        dag_name=f"{dag_name} - AJOUT",
+        run_name=run_name,
+        action_type=constants.SUGGESTION_SOURCE_AJOUT,
+    )
+    insert_suggestion(
+        df=df_acteur_to_delete,
+        metadata=metadata,
+        dag_name=f"{dag_name} - SUPRESSION",
+        run_name=run_name,
+        action_type=constants.SUGGESTION_SOURCE_SUPRESSION,
+    )
+    insert_suggestion(
+        df=df_acteur_to_update,
+        metadata=metadata,
+        dag_name=f"{dag_name} - MISES A JOUR",
+        run_name=run_name,
+        action_type=constants.SUGGESTION_SOURCE_MISESAJOUR,
+    )
+
+
+def insert_suggestion(
+    df: pd.DataFrame, metadata: dict, dag_name: str, run_name: str, action_type: str
+):
+    if df.empty:
+        return
+    engine = PostgresConnectionManager().engine
+    current_date = datetime.now()
+
+    with engine.connect() as conn:
+        # Insert a new suggestion
+        result = conn.execute(
+            """
+            INSERT INTO data_suggestioncohorte
+            (
+                identifiant_action,
+                identifiant_execution,
+                type_action,
+                statut,
+                metadata,
+                cree_le,
+                modifie_le
+            )
+            VALUES (%s, %s, %s, %s, %s, %s, %s)
+            RETURNING ID;
+        """,
+            (
+                dag_name,
+                run_name,
+                action_type,
+                constants.SUGGESTION_AVALIDER,
+                json.dumps(metadata),
+                current_date,
+                current_date,
+            ),
+        )
+        suggestion_cohorte_id = result.fetchone()[0]
+
```

Comment:
Voir si besoin ou pas de la sortir 

```suggestion
def prepare_cohorte_for_insertion(...):
    with engine.connect() as conn:
        # Insert a new suggestion
        result = conn.execute(
            """
            INSERT INTO data_suggestioncohorte
            (
                identifiant_action,
                identifiant_execution,
                type_action,
                statut,
                metadata,
                cree_le,
                modifie_le
            )
            VALUES (%s, %s, %s, %s, %s, %s, %s)
            RETURNING ID;
        """,
            (
                dag_name,
                run_name,
                action_type,
                constants.SUGGESTION_AVALIDER,
                json.dumps(metadata),
                current_date,
                current_date,
            ),
        )
        suggestion_cohorte_id = result.fetchone()[0]

```

pas bloquant

Reply:
A revoir avec l'engine Django